infrastructure/templates/public-api/components/containerApp.bicep

```diff
@@ -54,87 +51,39 @@ param minReplica int = 1
-
-@description('Specifies the subnet id')
-param subnetId string
+param appSettings {
+  name: string
+  @secure()
+  value: string
+}[]
 
 @description('A set of tags with which to tag the resource in Azure')
 param tagValues object
 
-@description('The Application Insights key that is associated with this resource')
-param applicationInsightsKey string
+@description('An existing Managed Identity\'s Resource Id with which to associate this Container App')
+param managedIdentityId string?
 
-var containerImageName = useDummyImage == true ? 'mcr.microsoft.com/azuredocs/aci-helloworld' : '${acrLoginServer}/${containerAppImageName}'
-var containerEnvName = '${resourcePrefix}-cae-${containerAppName}'
-var containerApplicationName = toLower('${resourcePrefix}-ca-${containerAppName}')
-var userIdentityName = '${resourcePrefix}-id-${containerAppName}'
-var containerLogName = '${resourcePrefix}-log-${containerAppName}'
-var acrPullRole = resourceId('Microsoft.Authorization/roleDefinitions', '7f951dda-4ed3-4680-a7ca-43fe172d538d')
+@description('Id of the owning Container App Environment')
+param managedEnvironmentId string
 
-resource logAnalytics 'Microsoft.OperationalInsights/workspaces@2021-06-01' = {
-  name: containerLogName
-  location: location
-  properties: {
-    sku: {
-      name: 'PerGB2018'
-    }
-  }
-  tags: tagValues
-}
-
-resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
-  name: userIdentityName
-  location: location
-}
-
-@description('This allows the managed identity of the container app to access the registry, note scope is applied to the wider ResourceGroup not the ACR')
-resource managedIdentityRBAC 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (!useDummyImage) {
-  name: guid(resourceGroup().id, managedIdentity.id, acrPullRole)
-  properties: {
-    roleDefinitionId: acrPullRole
-    principalId: managedIdentity.properties.principalId
-    principalType: 'ServicePrincipal'
-  }
-}
+var containerImageName = '${acrLoginServer}/${containerAppImageName}'
+var containerApplicationName = toLower('${resourcePrefix}-ca-${containerAppName}')
 
-resource containerAppEnvironment 'Microsoft.App/managedEnvironments@2023-05-01' = {
-  name: containerEnvName
-  location: location
-  properties: {
-    vnetConfiguration: {
-      infrastructureSubnetId: subnetId
-    }
-    daprAIInstrumentationKey: applicationInsightsKey
-    appLogsConfiguration: {
-      destination: 'log-analytics'
-      logAnalyticsConfiguration: {
-        customerId: logAnalytics.properties.customerId
-        sharedKey: logAnalytics.listKeys().primarySharedKey
-      }
-    }
-    workloadProfiles: [
-      {
-        name: 'Consumption'
-        workloadProfileType: 'Consumption'
-      }
-    ]
+var containerAppIdentity = managedIdentityId != null ? {
+  type: 'UserAssigned'
+  userAssignedIdentities: {
+    '${managedIdentityId}': {}
   }
-  tags: tagValues
+} : {
+  type: 'SystemAssigned'
 }
```

Comment:
Rather than declaring this, why don't you reference the existing resource which has to exist at this point?

```
resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = {
  name: identityName
}
```

I think there's a reason why this has to be a managed identity rather than a system assigned identity because of the way a container app throws a provision error if it doesn't have the ACR pull role in advance of the container app being provisioned.

Reply:
Yup that sounds like a good plan, thanks!  Updated as part of EES-4807.